crates/bevy_ui/src/render/mod.rs

```diff
@@ -984,6 +987,122 @@ pub fn extract_text_shadows(
+        Query<(
+            Entity,
+            &ComputedNode,
+            &ComputedNodeTarget,
+            &UiGlobalTransform,
+            &InheritedVisibility,
+            Option<&CalculatedClip>,
+            &TextLayoutInfo,
+            &TextOutline,
+        )>,
+    >,
+    camera_map: Extract<UiCameraMap>,
+) {
+    let mut start = extracted_uinodes.glyphs.len();
+    let mut end = start + 1;
+
+    let mut camera_mapper = camera_map.get_mapper();
+    for (
+        entity,
+        uinode,
+        target,
+        global_transform,
+        inherited_visibility,
+        clip,
+        text_layout_info,
+        outline,
+    ) in &uinode_query
+    {
+        // Skip if not visible or if size is set to zero (e.g. when a parent is set to `Display::None`)
+        if !inherited_visibility.get() || uinode.is_empty() {
+            continue;
+        }
+
+        let Some(extracted_camera_entity) = camera_mapper.map(target) else {
+            continue;
+        };
+
+        let width = outline.width as i32;
+        let aa_factor = outline.anti_aliasing.unwrap_or(1.0);
+        let color: LinearRgba = outline.color.into();
+
+        for (
+            i,
+            PositionedGlyph {
+                position,
+                atlas_info,
+                span_index,
+                ..
+            },
+        ) in text_layout_info.glyphs.iter().enumerate()
+        {
+            let rect = texture_atlases
+                .get(&atlas_info.texture_atlas)
+                .unwrap()
+                .textures[atlas_info.location.glyph_index]
+                .as_rect();
+
+            for offset_x in -width..=width {
+                for offset_y in -width..=width {
+                    if offset_x == 0 && offset_y == 0 {
+                        continue;
+                    }
+
+                    // Anti-aliasing.
+                    let mut color = color;
+                    if (offset_x.abs() == width) || (offset_y.abs() == width) {
+                        color.alpha *= aa_factor;
+                    }
+
+                    let offset = Vec2 {
+                        x: offset_x as f32,
+                        y: offset_y as f32,
+                    };
+
+                    let transform = Affine2::from(*global_transform)
+                        * Affine2::from_translation(
+                            -0.5 * uinode.size() + offset / uinode.inverse_scale_factor(),
+                        );
+
+                    extracted_uinodes.glyphs.push(ExtractedGlyph {
+                        transform: Affine2::from_mat3(
+                            transform * Mat3::from_translation(*position),
+                        ),
+                        rect,
+                    });
+
+                    if text_layout_info.glyphs.get(i + 1).is_none_or(|info| {
+                        info.span_index != *span_index
+                            || info.atlas_info.texture != atlas_info.texture
+                    }) {
```

Comment:
```suggestion
                    if text_layout_info.glyphs.get(i + 1).is_none_or(|info| {
                        info.atlas_info.texture != atlas_info.texture
                    }) {
```
I think it's safe to remove this, since span-bound text color is irrelevant. @ickshonpe what do you think?

Reply:
Also, this `ExtractedUiNode` bit needs to be moved to the outer loop.

Reply:
If i move the extraced_uinodes.push bit, I get strange artifacts in the output...

Reply:
> If i move the extraced_uinodes.push bit, I get strange artifacts in the output...

Only `extracted_uinodes.uinodes` should be in the outer loop (after the inner loop). The `.glyphs` needs to stay in the inner loop.

Reply:
Ah looks like `end += 1;` needs to be moved to the inner loop after pushing to `.glyphs`.

Reply:
Ok I fixed it in `bevy_cobweb_ui`: <https://github.com/UkoeHB/bevy_cobweb_ui/blob/a3265e1c80a4b21cdd7cf8200e294f2a6712248e/src/ui_bevy/ui_ext/text_rendering.rs#L11>

Reply:
> I think it's safe to remove this, since span-bound text color is irrelevant. @ickshonpe what do you think?

Yep not needed.